Fixes for unicycler

davebx opened the pull request:
This should resolve the failing tests on https://github.com/galaxyproject/tools-iuc/pull/1129

shiltemann committed:
filter.seqs small fixes

shiltemann committed:
classify.seqs: allow providing names file

shiltemann committed:
remove.seqs: handle empty accnos files

shiltemann committed:
rarefaction.single: fix dataset discovery pattern

shiltemann committed:
rarefaction.single: nicer element names

shiltemann committed:
rarefaction.single: nicer calculator names

shiltemann committed:
chimera.uchime: fix wrong filename

shiltemann committed:
sub.sample: fix taxonomy datatype

shiltemann committed:
unique.seqs: allow mothur.align as well as fasta

shiltemann committed:
sub.sample: add taxonomy output

shiltemann committed:
pcr.seqs: stricter output names

shiltemann committed:
make.group: support creation of group file from collection

shiltemann committed:
count.seqs: fix group select

shiltemann committed:
screen.seqs: update help text

shiltemann committed:
merge.files: update options text

shiltemann committed:
add toolshed category

shiltemann committed:
filter.seqs: only output collection if multiple outputs were created

shiltemann committed:
unique.seqs: allow count_table input

shiltemann committed:
pre.cluster: add support for count file and penalties

shiltemann committed:
pre.cluster: update filter

shiltemann committed:
align.seqs: fix default penalties

shiltemann committed:
screen.seqs: fix numeric check

shiltemann committed:
unique.seqs: fix filters and combine name and count inputs

shiltemann committed:
screen.seqs: fix output datatype and update tests

shiltemann committed:
chimera.uchime: add count table output and allow fasta input

shiltemann committed:
seq.error: allow nonaligned fastas and remove output picker

shiltemann committed:
add different outputs for group or count file input

shiltemann committed:
cluster: names file should be optional

shiltemann committed:
dist.seqs: allow unaligned fasta input

shiltemann committed:
remove.groups: fix filters

shiltemann committed:
cluster.split: some input files are optional

shiltemann committed:
add stdio macro for better error detection

shiltemann committed:
fix macro import

shiltemann committed:
stricter error detection

shiltemann committed:
make.shared: fix filters

shiltemann committed:
move stdio before version_command

shiltemann committed:
allow mothur.dist datatype

shiltemann committed:
allow mothur.dist datatype for phylip files

shiltemann committed:
fix/remove broken URLs

shiltemann committed:
get.mimarkspackage: remove invalid option

shiltemann committed:
mimarks.attributes: fix documentation

shiltemann committed:
parsimony: fix test

shiltemann committed:
pcr.seqs: fix tests

shiltemann committed:
remove.otulabels: fix label parameter

shiltemann committed:
trim.seqs: fix test

shiltemann committed:
unique.seqs: fix test

shiltemann committed:
*.flows: update tests

shiltemann committed:
use tee to make logfile name predictable

shiltemann committed:
oops

shiltemann committed:
get.mimarkspackage: remove file option

shiltemann committed:
move calculators to macro

shiltemann committed:
new tool: get.communitytype

shiltemann committed:
get.communitytype: add test

shiltemann committed:
I promise I know how to spell..

shiltemann committed:
new tool: classify.rf

shiltemann committed:
add testdata for classify.rf

shiltemann committed:
new tool: lefse

shiltemann committed:
fix default setting for gapopen

shiltemann committed:
amova: filter on mothur.design format

shiltemann committed:
make name file optional

nekrut commented:
For testing I was using unicycler outside conda. The unicycler has been updated recently, so perhaps we need to update the conda package, no?

bgruening commented:
Try the conda version `0.3.0b`.

nekrut commented:
@bgruening: I do not see `0.3.0b` in bioconda though

martenson commented:
@bgruening @nekrut https://bioconda.github.io/recipes.html and https://anaconda.org/bioconda/unicycler are giving different results :/ version 0.3.0b is in the latter...

shiltemann committed:
set pipefail option to fix lost exit codes due to using tee

shiltemann committed:
get.communitytype: fix test
test was occasionaly failing due to stochastic processes in the algorithm

shiltemann committed:
make.contigs: allow choice between collection or regular input

arbernard committed:
Align.seqs - gapextend default value update

The Help of Mothur 1.36.1 says that the default value for the align.seqs gapextend parameter is -2.

arbernard committed:
Align.seqs - Optional accnos output file recovery

When the flip parameter is set to yes then Mothur align.seqs can produce an extra output file in mothur.accnos format. The wrapper is now able to recover it when necessary.

arbernard committed:
Align.seqs - Addition of a test for the accnos file recovery

An additional test has been added to verify that the wrapper is capable of collecting an accnos file when needed.

arbernard committed:
Cluster.split - Merge of the names and count parameters

In a similar way of what has been done in the pre.cluster tool, end users can now choose to use either a name file (mothur.names) or a count table (mothur.count_table) with a single parameter. The inputs section now have a nameOrCount param and in the command section the appropriate command line option (name or count) is selected automatically base on the format of the selected file.

arbernard committed:
Cluster.split - Count input removal

The count input param is now properly removed.

arbernard committed:
Cluster.split - No fasta parameter in the Classify when case

Since there is a fasta when case in the SplitBy conditional, there is no use for a fasta parameter in the Classify when case.

arbernard committed:
Cluster.split - Mandatory parameters

Some parameters where improperly considered as optionals. The name/count file is only optional in Classify-Phylip mode and mandatory in other modes. The dist and taxonomy files are mandatory in Classify mode. The Fasta and taxonomy files are mandatory in Fasta mode.

arbernard committed:
Cluster.split - Sabund/Rabund output files filtering

The sabund and rabund files are not generated when a mothur.count_table file is used instead of a mothur.names file. As a consequence, filters have been added to avoid empty output files in the histories.

arbernard committed:
Cluster.split - Fix minimum value check for taxlevel parameter

The minimum allowed value for the taxlevel parameter is now defined with the "min" attribute of the input param itself. It also fix the issue where the default value was ignored with the previous syntax (when default value was set to 1).

arbernard committed:
Cluster.split - Update of the tests section

The existing tests of the tests section have been updated to reflect the changes made in the previous commits. A new test to check the proper execution of the tool with a mothur.count_table file have also been added.

arbernard committed:
Get.oturep - Merge of the names and count parameters

In a similar way of what has been done in the pre.cluster tool, end users can now choose to use either a name file (mothur.names) or a count table (mothur.count_table) with a single parameter. The inputs section now have a nameOrCount param and all other sections have been updated accordingly. In the command section the appropriate command line option (name or count) is selected automatically base on the format of the selected file. In addition, a name/count file is now considered as mandatory in column mode (as stated in get.oturep(help)).

arbernard committed:
Get.oturep - Selection method switch

In the previous versions of the wrapper, the switch between the two available methods for representative sequences selection was considered as a secondary parameter. However, it should be consider as a main switch because most other options depends on it. This have been changed in this version and every section's content have been adapted.

arbernard committed:
Get.oturep - New test for abundance mode

A new test designed to check the appropriate execution of the tool in abundance mode has been added.

arbernard committed:
Create.database - Merge of the repname and count parameters

In a similar way of what has been done in the pre.cluster tool, end users can now choose to use either a name file (mothur.names) or a count table (mothur.count_table) with a single parameter. The inputs section now have a nameOrCount param and in the command section the appropriate command line option (repname or count) is selected automatically base on the format of the selected file. In addition, the order of the parameters in every sections has been homogenized.

arbernard committed:
Create.database - New test and test file

A new test has been added to the tests section and the corresponding mothur.count_table input file has been added in the test-data folder.

arbernard committed:
Make.contigs - New possible input type (list:paired collection)

The end user is now able to choose between standalone forward and reverse fastq files, a simple "paired" collection (one pair) or a list:paired collection (multiple pairs).

arbernard committed:
Make.contigs - Update of the command section with combo file creation

The "command" section has been updated to match the new input_type param possible values. When the end user select a list:paired collection as input then a combo file is created on the fly and the execution of make.contigs is switched to combo mode.

arbernard committed:
Make.contigs - Update of the outputs section

Since the name of the output files are based on the name of either the "ffastq" or "file" option of Mothur make.contigs then the value of the from_work_dir attributes must be slightly changed to match both cases. As a consequence, Galaxy is now looking for files matching the *fastq* pattern instead of the ffastq* pattern (The combo file is called combo_fastq.dat).

arbernard committed:
Make.contigs - New test for list:paired collections

A new <test> has been added in the tests section to be able to test the list:paired collection case with planemo. In addition, the syntax of the conditional params has been made more explicite and the descriptions of the tests have been slightly modified to become a bit clearer.

shiltemann committed:
hcluster: fix test

shiltemann committed:
use https links

bebatut committed:
Start to write the data manager for qiime

bebatut committed:
Fix db extraction

bebatut committed:
Change the version for silva db

bebatut committed:
Add missing tables

bebatut committed:
Remove output.json (test)

shiltemann committed:
remove tool_dependencies.xml

shiltemann committed:
hcluster: fix test

shiltemann committed:
add instructions for installation via API

shiltemann committed:
fix version command

shiltemann committed:
fix url

bebatut committed:
Fix planemo linting

bebatut committed:
Fix download for data

bebatut committed:
Add missing sample tool data

bebatut committed:
Change move type

bebatut committed:
Add missing data table in conf

bebatut committed:
Remove some not working options

bebatut committed:
Fix some path issues

bgruening committed:
change travis tests to split by tools not by repos

bgruening committed:
don't lint, I will put this back promised

bgruening committed:
remove string

bgruening committed:
travis work

bgruening committed:
yaml fix

bgruening committed:
use repositories for linting

bgruening committed:
set MPLBACKEND

bgruening committed:
MPLBACKEND="Agg"

blankclemens committed:
Simplify replicates logic.

shiltemann committed:
fix mothur version

bgruening committed:
Merge pull request #1303 from blankclemens/masigpro

Add maSigPro wrapper

shiltemann committed:
makd_emperor: fix test data paths

shiltemann committed:
Merge branch 'qiime' into qiime

bgruening committed:
Change it back.

nsoranzo committed:
Fixes

bgruening committed:
Update .shed.yml

bgruening committed:
Merge pull request #1265 from wjurkowski/qiime

draft of pick_closed_reference_otus

nsoranzo committed:
Temporarily revert change which breaks Travis

nsoranzo committed:
Merge pull request #1229 from galaxyproject/travis_for_dm_pkg

consider packages and data-managers as well

bgruening committed:
Merge branch 'master' into qiime

shiltemann committed:
align_seqs: fix tests

shiltemann committed:
Merge branch 'qiime' of github.com:galaxyproject/tools-iuc into qiime

shiltemann committed:
oops

lecorguille committed:
Update prinseq.xml

To keep the `fastqsanger` format through the tool

bgruening committed:
fix shed.yaml

bgruening committed:
pep8

lecorguille committed:
Add ftype in test

bgruening committed:
more pep8

nsoranzo committed:
Merge pull request #1307 from lecorguille/patch-1

Update prinseq.xml

shiltemann committed:
update travis to chunk by tool instead of repo
all tools in a changed repo will still be tested, even if tool xml itself was not changed

bgruening committed:
Merge pull request #1306 from galaxyproject/qiime_data_manager_2

QIIME - Add data manager

shiltemann committed:
fix test

shiltemann committed:
fix test filter_otus_from_otu_table

shiltemann committed:
fix test for multiple_join_paired_ends

shiltemann committed:
exclude data managers and packages from testing

shiltemann committed:
comment out hanging test

shiltemann committed:
split_librarires: fix test

shiltemann committed:
upgma_cluster: fix output naming and test

bebatut committed:
Update the data table to fit the data manager ones

bebatut committed:
Use the correct table for pick_otus

bebatut committed:
Remove extra lines

bebatut committed:
pick_open_reference_otus: Add link to correct db table and rename some param

bebatut committed:
align_seqs: remove extra lines

bebatut committed:
alpha_diversity: remove extra lines

bebatut committed:
validate_mapping_file: remove extra lines

bebatut committed:
summarize_taxa: remove extra lines

bebatut committed:
split_libraries_fastq: remove extra lines

bebatut committed:
split_libraries: remove extra lines

bebatut committed:
make_emperor: remove extra lines

bebatut committed:
jackknifed_beta_diversity: remove extra lines

bebatut committed:
filter_otus_from_otu_table: remove extra lines

bebatut committed:
count_seqs: remove extra lines

bebatut committed:
core_diversity_analyses: remove extra lines and spaces

bebatut committed:
compare_categories: remove extra lines

bebatut committed:
beta_diversity_through_plots: remove extra lines

bebatut committed:
beta_diversity: remove extra lines

bebatut committed:
assign_taxonomy: remove extra lines

bebatut committed:
alpha_rarefaction: remove extra lines

shiltemann committed:
fix invalid DISPLAY variable?

bebatut committed:
compare_categories: add r requirement

bebatut committed:
Update the .shed.yml for suite

bebatut committed:
Reorganize in alphabetical order the script to generate test data

shiltemann committed:
fix colons in yaml

shiltemann committed:
use both

shiltemann committed:
split libraries: switch to assert_contents

shiltemann committed:
compare_categories: use r version that plays nice with both r-vegan and r-optparse

shiltemann committed:
split_libraries: fix test

shiltemann committed:
fix dependency order

shiltemann committed:
core_diversity_analysis: split test in two to avoid travis timeout

bebatut committed:
core_diversity_analyses: split again the test

bebatut committed:
core_diversity_analyses: simplify the alpha diversity test

bgruening committed:
Merge pull request #1302 from loraine-gueguen/trinity_analyze_diff_expr_fix_input_files_name

Trinity analyze_diff_expr: fix input file names

bebatut committed:
Add travis wait

bebatut committed:
core_diversity_analyses: comment alpha diversity test

bebatut committed:
Remove the travis wait

bebatut committed:
core_diversity_analyses: remove alpha diversity test

shiltemann committed:
make.contigs: output group file in combo mode

shiltemann committed:
assign_taxonomy: fix filter

shiltemann committed:
fix biom format

hexylena committed:
Merge pull request #431 from galaxyproject/qiime

Add Qiime wrappers

shiltemann committed:
remove broken link and add paper as citation

bgruening committed:
Merge pull request #967 from galaxyproject/mothur-fixes

Mothur fixes

bgruening committed:
activate pep8 testing again

bgruening committed:
change id

bgruening committed:
change id

bgruening committed:
change id

bgruening committed:
change id

bgruening committed:
safeguard the tool search

shiltemann committed:
reenable alpha diversity test

nsoranzo committed:
Run planemo test only if changed_tools_chunk.list exists

bgruening committed:
Merge pull request #1314 from galaxyproject/qiime-tests

QIIME: re-enable alpha diversity test

nsoranzo committed:
Merge pull request #1311 from galaxyproject/pep8

activate pep8 testing again

Commit:
small tweaks

nsoranzo committed:
Check that changed_tools_chunk.list is also not empty

It may be empty in some chunk.
Also:
- rename changed_repositories_chunk.list to changed_repositories.list
- run `planemo conda_install` only if `planemo test` will be run

jj-umn committed:
bcftools concat - add naive option

naive option needed for merging vcf files generated per chromosome:
HISAT2 -> bamtools split -reference => bcftools mpileup => bcftools
call => bcftools concat (naive) ->

jj-umn committed:
bcftools concat fix command cheetah

fixed rm_dups

tiagoantao committed:
support for single end reads

tiagoantao committed:
support for single end reads

bgruening committed:
Merge pull request #1316 from nsoranzo/travis

Check that changed_tools_chunk.list is also not empty

jj-umn committed:
bcftools concat naive

Added test for naive option.
Added bcftools index when converting to uncompressed vcd output.

tiagoantao committed:
use of optional parameters

tiagoantao committed:
all test files gzipped and output always gzipped

tiagoantao committed:
forgot test file

bgruening committed:
Merge pull request #1291 from tiagoantao/clone_filter

STACKS: support for clone_filter

bgruening committed:
Delete tool_dependencies.xml

bgruening committed:
small fixes and version bump

tiagoantao committed:
fastqcsangergz -> fastqcsanger.gz

peterjc committed:
Merge pull request #1324 from tiagoantao/clone_filter

Correcting a bug on the fastqsanger.gz output datatype

bgruening committed:
fit req to conda version

mvdbeek committed:
Merge pull request #1319 from galaxyproject/condafy_column_join

Delete tool_dependencies.xml from column_join

davebx committed:
Upgrade freebayes to 1.1.0, remove multiple from bam input.

davebx committed:
Allow both proceesing modes for multiple inputs.

davebx committed:
Fix linting error.

jj-umn committed:
Merge pull request #1317 from galaxyproject/bcftools_concat_naive

bcftools concat - add naive option

rhpvorderman committed:
Fixed infinite loop error in data_manager_gatk_picard_index_builder

bgruening committed:
few changes

bebatut committed:
Add missing tools to biom-format (#1329)

* Add biom_from_uc tool

* biom_add_metadata: use param's argument instead of name

* biom_convert: use param's argument instead of name

* biom_normalize_table: start wrapper

* biom_subset_table: start wrapper

* biom_summarize_table: start wrapper

* biom_convert: rename variables to remove -

* biom_add_metadata: rename variables to remove -

* biom_normalize_table: rename variables to remove -

* biom_subset_table: rename variables to remove - and fix tests

* biom_summarize_table: rename variables to remove - and fix tests

* biom_from_uc: rename variables to remove - and fix tests

* Remove extra space and increment version

* Replace the stdio by detect_errors

* Change tabular to biom1 for the input

* Change tabular to txt for input

* Downsample the uc test data

* Change biom2 to h5

* fix output name in tests

bgruening committed:
Merge pull request #1331 from rhpvorderman/fix_picard_index_builder

Fixed infinite loop error in data_manager_gatk_picard_index_builder

nsoranzo committed:
Condafy TransDecoder. Update to v3.0.1

mvdbeek committed:
Fix arguement typo in bowtie2

blankclemens committed:
Fix bug in stepback.R function.
Update maSigPro package to 1.49.3

blankclemens committed:
Update test file.

bgruening committed:
Merge pull request #1335 from nsoranzo/transdecoder

Condafy TransDecoder. Update to v3.0.1

mvdbeek committed:
Merge pull request #1300 from nekrut/bowtie2.3.2

transfer bowtie2 from devteam and add interleaved support

nsoranzo committed:
Add 'merge' to stringtie_merge tool name

Fix https://github.com/galaxyproject/tools-iuc/issues/1320 .
Also:
- Fix linting warnings
- Use single quotes in `<command>`
- Fix missing param labels in stringtie_merge
- Remove redundant option explanations in `<help>`
- Reorder XML attributes according to IUC coding style

mvdbeek committed:
Allow fasta input in bowtie2

and bump version.

bgruening committed:
Merge pull request #1336 from blankclemens/masigpro_update

maSigPro wrapper update

bgruening committed:
Merge pull request #1337 from mvdbeek/bowtie2_fasta_input

Allow fasta input in bowtie2

bgruening committed:
Merge pull request #1339 from nsoranzo/fix_1320

Add 'merge' to stringtie_merge tool name

tiagoantao committed:
corrects a bug with barcoding

bgruening committed:
Merge pull request #1340 from tiagoantao/clone_filter

corrects a bug with barcoding

mvdbeek committed:
Use element identifier when building report

mvdbeek committed:
Escape the element_identifier (thx @jmchilton)

nsoranzo committed:
Condafy seqtk

Remove tool_dependencies.xml

Also:
- Fix seqtk_seq `-M` param to be of type `data`
- Single quotes around data and text params in `<command>`
- Consistent 4-spaces indentation
- Use `argument` attribute

mvdbeek committed:
Merge pull request #1342 from nsoranzo/seqtk

Condafy seqtk

bgruening committed:
Merge pull request #1341 from mvdbeek/fastqc_element_identifier

Use element identifier when building report

davebx committed:
Remove problematic line from test data.

bgruening committed:
Merge pull request #1328 from davebx/freebayes

Upgrade and update FreeBayes

davebx committed:
Fixes for unicycler

nsoranzo committed:
Use FreeBayes 1.1.0 also for leftalign

- Add `requirements` macro
- Make genotype_variant_threshold param optional
- Use `detect_errors="exit_code"`
- Fix argument for `e` param
- Use `--will-cite` option of GNU parallel instead of deprecated `--no-notice`
- Remove unnecessary quotes in command
- Remove redundant help
- IUC code style fixes

bgruening committed:
Merge pull request #1344 from nsoranzo/freebayes

Use FreeBayes 1.1.0 also for leftalign

bgruening committed:
remove tool dependencies

bgruening committed:
upgrade to conda dependencies

bgruening committed:
remove tool from blacklist

bgruening committed:
This looks ugly ;)

nsoranzo committed:
Merge pull request #1346 from galaxyproject/trans

transtermhp goes conda

nsoranzo committed:
Deprecate gatk2, rglasso and tool_factory_2

bgruening committed:
Merge pull request #1347 from nsoranzo/deprecate_tools

Deprecate gatk2, rglasso and tool_factory_2

davebx committed:
Merge branch 'master' of git://github.com/galaxyproject/tools-iuc into unicycler
